{afd} Fd api version update to 2024

Ptnan7 opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Commit:
auto gen

Commit:
auto gen

HarveyLink committed:
Merge remote-tracking branch 'upstream/main' into pr/42502

HarveyLink committed:
regen

HarveyLink committed:
Regen

HarveyLink committed:
Fix ModelFactory

HarveyLink committed:
Update Azure.ResourceManager.Cdn.netstandard2.0.cs

Ptnan7 committed:
Merge branch 'main' into main

Commit:
add codeowner

Commit:
add test

Commit:
test done

Ptnan7 committed:
Merge branch 'Azure:main' into main

Commit:
gen 2024

Commit:
update test

Commit:
update

Commit:
pass test

Commit:
gen 2024 api

Commit:
code owner change

Commit:
code owner

Commit:
code owner

Commit:
change api version

Commit:
update to 2024

Commit:
gen api

Commit:
add code owner

Commit:
update

pshao25 committed:
Update communication to latest spec (#42986)

* Update communication to latest spec

* update

* update

RodgeFu committed:
Update comment of SetApiVersion to include more information about the ResourceType (#43062)

* update comment of SetApiVersion

* Update sdk/resourcemanager/Azure.ResourceManager/src/ArmClientOptions.cs

Co-authored-by: Dapeng Zhang <ufo54153@gmail.com>

* move to param doc

---------

Co-authored-by: Dapeng Zhang <ufo54153@gmail.com>

ArthurMa1978 committed:
Polish HealthcareApis and CosmosDB (#43089)

hsurana06 committed:
Upgrading SDK for Microsoft.SCVMM with stable swagger API version 2023-10-07 (#39493)

pshao25 committed:
Update healthinsights spec to latest (#43064)

* update commit in healthinsights

* update

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240329.1 (#43091)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.PostgreSql (#43086)

azure-sdk committed:
Increment package version after release of Azure.Provisioning (#43079)

jsquire commented on .github/CODEOWNERS:
```diff
@@ -930,6 +927,12 @@
 # PRLabel: %Network - CDN
 /sdk/cdn/                                                          @ptnan7
 
+# ServiceLabel: %Network - Front Door %Mgmt
```
Please move this section down so that alphabetical ordering is preserved.   It should go around L945.

jsquire commented on sdk/cdn/Azure.ResourceManager.Cdn/assets.json:
If this PR is scoped to Front Door, there shouldn't be changes to CDN included.   Please revert these or move them to a new PR.

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ScVmm (#43096)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7977 (#43094)

* Skip API review package name approval for management plane packages

jsquire committed:
[CODEOWNERS] Fix linter (#43097)

Fixing error message for an account that recently had write permission granted.

benbp committed:
Add 1es-redirect template. Support template validation canary runs (#43049)

azure-sdk committed:
Increment package version after release of Azure.Security.CodeTransparency (#43098)

azure-sdk committed:
Increment package version after release of Azure.Monitor.Query (#43104)

Commit:
Merge branch 'fd-2024' of https://github.com/Ptnan7/azure-sdk-for-net into fd-2024
